tests/MongoDB.Driver.Tests/EncryptionTests.cs

```diff
@@ -0,0 +1,174 @@
+        #region static
+        private static readonly CollectionNamespace __keyVaultCollectionNamespace = CollectionNamespace.FromFullName("db.coll");
+        #endregion
+
+        private const string LocalMasterKey = "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk";
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void CryptClient_should_be_initialized([Values(false, true)] bool withAutoEncryption)
+        {
+            RequireServer.Check().Supports(Feature.ClientSideEncryption);
+
+            using (var client1 = GetClient(withAutoEncryption))
+            using (var client2 = GetClient(withAutoEncryption))
+            {
+                var libMongoCryptController1 = ((MongoClient)client1.Wrapped).LibMongoCryptController;
+                var libMongoCryptController2 = ((MongoClient)client2.Wrapped).LibMongoCryptController;
+                if (withAutoEncryption)
+                {
+                    var cryptClient1 = libMongoCryptController1._cryptClient();
+                    var cryptClient2 = libMongoCryptController2._cryptClient();
+                    var areTheSame = object.ReferenceEquals(cryptClient1, cryptClient2);
+                    areTheSame.Should().BeTrue();
+                }
+                else
+                {
+                    libMongoCryptController1.Should().BeNull();
+                    libMongoCryptController2.Should().BeNull();
+                }
+
+                var kmsProviders = GetKmsProviders();
+                var clientEncryption = GetClientEncryption((MongoClient)client1.Wrapped, kmsProviders);
+                clientEncryption._libMongoCryptController()._cryptClient().Should().NotBeNull();
+            }
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Mongocryptd_should_be_initialized_for_only_for_auto_encryption([Values(false, true)] bool withAutoEncryption)
+        {
+            RequireServer.Check().Supports(Feature.ClientSideEncryption);
+
+            using (var disposableClient = GetClient(withAutoEncryption))
+            {
+                var kmsProviders = GetKmsProviders();
+                var client = (MongoClient)disposableClient.Wrapped;
+                var clientEncryption = GetClientEncryption(client, kmsProviders);
+                if (withAutoEncryption)
+                {
+                    client.LibMongoCryptController.Should().NotBeNull();
+                    var clientController = client.LibMongoCryptController;
+                    clientController._mongocryptdClient().Should().NotBeNull();
+
+                    var clientEncryptionController = clientEncryption._libMongoCryptController();
+                    clientEncryptionController.Should().NotBeNull();
+                    var mongoCryptD = clientEncryptionController._mongocryptdClient();
+                    mongoCryptD.Should().BeNull();
```

Comment:
Remove lines 88-89.

`ExplicitEncryptionLibMongoCryptController` does not have a `_mongoCryptDClient` field.